fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java

```diff
@@ -946,8 +949,19 @@ public void modifyPartitionsProperty(Database db,
             if (tTabletType != partitionInfo.getTabletType(partition.getId())) {
                 partitionInfo.setTabletType(partition.getId(), tTabletType);
             }
+            // 5. enable data cache
+            if (olapTable.isCloudNativeTable()) {
+                DataCacheInfo dataCacheInfo = partitionInfo.getDataCacheInfo(partition.getId());
+                if (dataCacheInfo == null || newEnableDataCache != dataCacheInfo.isEnabled()) {
+                    GlobalStateMgr.getCurrentState().getStarOSAgent().updateShardGroup(
```

Comment:
may need to keep this change as close as the ModifyPartitionInfo persistency, otherwise the metadata can get easily out of sync, for example, the updateShardGroup is success, but this modification is failed at last. The system is in an inconsistent state.

Reply:
OK, I'll keep them as close as possible.